lib/DefaultResolver.php

```diff
@@ -214,6 +218,8 @@ private function doResolve($name, array $types, $options) {
             $this->config = (yield \Amp\resolve($this->loadResolvConf()));
         }
 
+        $config = $this->loadEnvConfig();
```

Comment:
We can provide an additional way to override the default config later, but I don't agree with you taking the `getenv` value the first time DNS is used only, because that might result in a behavior change if you use `setenv` and then introduce something that needs DNS before that.